src/garage/tf/algos/npo.py

```diff
@@ -607,6 +639,39 @@ def _fit_baseline(self, samples_data):
         else:
             self.baseline.fit(paths)
 
+    def _get_advantages(self, baseline_var, reward_var, name):
```

Comment:
what is the purpose of these two helper functions? they just call single other functions.

Reply:
The "calculate returns in separate rollouts" version of NPO will need to override this method.

Reply:
this version of NPO calculates returns in separate rollouts....

calculating returns across rollouts is unique to RL2 -- whatever is overriding should be in there. it makes no sense to make a special version of NPO which will only be used by RL2.

Reply:
Do you think we should create a new class RL2NPO without inheriting NPO?

Reply:
if you are asking me if you should copy/paste, no.

if you can define a class RL2 which uses composition instead of inheritance for NPO, and then inherit RL2PPO and RL2TRPO from that, then yes.

take a look at Keren's MAML for what this looks like.

Reply:
I can't use most of the code from NPO if I do composition. NPO builds the policy loss in a way that is not applicable to RL2. In order to implement "calculate returns with respect to each separate rollouts in a meta batch",  I think I can only reuse most of the code from NPO only if I inherit it and override some of the functions, especially the way it computes returns and advantages.

Reply:
@krzentner What do you think?

Reply:
I think it's fine to have these hook point methods. Admittedly, it would be better to do something similar to MAML, but that's more difficult to do in this case while still allowing us to support variations like concatenated paths, since the related logic is spread throughout the algorithm.